src/h3/mod.rs

```diff
@@ -25,6 +25,275 @@
+//!         # return;
+//!     },
+//!
+//!     Err(e) => {
+//!         // An error occurred, handle it.
+//!         # return;
+//!     },
+//! }
+//! ```
+//!
+//! ## Handling responses
+//!
+//! An HTTP/3 client uses [`poll()`] to read responses:
+//!
+//! ```no_run
+//! # let mut config = quiche::Config::new(quiche::VERSION_DRAFT18).unwrap();
+//! # config.set_application_protos(b"\x05h3-18").unwrap();
+//! # let server_name = "quic.tech";
+//! # let scid = [0xba; 16];
+//! # let h3_config = quiche::h3::Config::new(0, 1024, 0, 0).unwrap();
+//! # let mut conn = quiche::connect(Some(&server_name), &scid, &mut config).unwrap();
+//! # let mut h3_conn = quiche::h3::Connection::with_transport(&mut conn,
+//! #                                                       &h3_config).unwrap();
+//! match h3_conn.poll(&mut conn) {
+//!     Ok((stream_id, quiche::h3::Event::Headers(headers))) => {
+//!         let mut status = "";
+//!
+//!         for hdr in &headers {
+//!             match hdr.name() {
+//!                 ":status" => {
+//!                     status = hdr.value();
+//!                 },
+//!
+//!                 _ => (),
+//!             }
+//!         }
+//!         # return
+//!     },
+//!
+//!     Ok((stream_id, quiche::h3::Event::Data(data))) => {
+//!         print!("Received {} bytes of payload on stream {}", data.len(), stream_id);
+//!         # return;
+//!     },
+//!
+//!     Err(quiche::h3::Error::Done) => {
+//!         // Done reading.
+//!         # return;
+//!     },
+//!
+//!     Err(e) => {
+//!         // An error occurred, handle it.
+//!         # return;
+//!     },
+//! }
+//! ```
+//!
+//! HTTP/3 request and response exchanges may consist of several HEADERS and
+//! DATA frames. Calling [`poll()`] repeatedly will generate an [`Event`] for
+//! each. The end of the exchange can be determined by using the QUIC connection
+//! [`stream_finished()`] function:
+//!
+//! ```no_run
+//! # let mut config = quiche::Config::new(quiche::VERSION_DRAFT18).unwrap();
+//! # config.set_application_protos(b"\x05h3-18").unwrap();
+//! # let server_name = "quic.tech";
+//! # let scid = [0xba; 16];
+//! # let h3_config = quiche::h3::Config::new(0, 1024, 0, 0).unwrap();
+//! # let mut conn = quiche::connect(Some(&server_name), &scid, &mut config).unwrap();
+//! # let mut h3_conn = quiche::h3::Connection::with_transport(&mut conn,
+//! #                                                       &h3_config).unwrap();
+//! match h3_conn.poll(&mut conn) {
+//!     Ok((stream_id, quiche::h3::Event::Headers(headers))) => {
+//!         let mut status = "";
+//!
+//!         for hdr in &headers {
+//!             match hdr.name() {
+//!                 _ => (),
+//!             }
+//!         }
+//!         # return
+//!     },
+//!
+//!     Ok((stream_id, quiche::h3::Event::Data(data))) => {
+//!         print!("Received {} bytes of payload on stream {}", data.len(), stream_id);
+//!
+//!         if conn.stream_finished(stream_id) {
```

Comment:
I've just realized we are probably doing this wrong. The check should go after the match.

Anyway, for the purposes of this example we should probably avoid repeating the whole match becase it's kind of hard to see the difference from the previous one.